test/api/visualization/annotations_controller_test.rb

```diff
@@ -0,0 +1,98 @@
+require 'api_test_helper'
+require 'test_instrumentor'
+
+class AnnotationsControllerTest < ActionDispatch::IntegrationTest
+  include Devise::Test::IntegrationHelpers
+  include Requests::JsonHelpers
+  include Requests::HttpHelpers
+  include Minitest::Hooks
+    include TestInstrumentor
+
+  before(:all) do
+    @user = FactoryBot.create(:api_user)
+    OmniAuth.config.mock_auth[:google_oauth2] = OmniAuth::AuthHash.new({
+                                                                           :provider => 'google_oauth2',
+                                                                           :uid => '123545',
+                                                                           :email => 'testing.user@gmail.com'
+                                                                       })
+    @basic_study = FactoryBot.create(:detached_study,
+                                     name: 'Basic Cluster Study',
+                                     public: false,
+                                     user: @user)
+    @basic_study_cluster_file = FactoryBot.create(:cluster_file,
+                                                  name: 'clusterA.txt',
+                                                  study: @basic_study,
+                                                  cell_input: {
+                                                     x: [1, 4 ,6],
+                                                     y: [7, 5, 3],
+                                                     cells: ['A', 'B', 'C']
+                                                  },
+                                                  annotation_input: [{name: 'foo', type: 'group', values: ['bar', 'bar', 'baz']}])
+
+    @basic_study_metadata_file = FactoryBot.create(:metadata_file,
+                                                   name: 'metadata.txt',
+                                                   study: @basic_study,
+                                                   cell_input: ['A', 'B', 'C'],
+                                                   annotation_input: [
+                                                     {name: 'species', type: 'group', values: ['dog', 'cat', 'dog']},
+                                                     {name: 'disease', type: 'group', values: ['none', 'none', 'measles']}
+                                                   ])
+    @user2 = FactoryBot.create(:api_user)
+    @empty_study = FactoryBot.create(:detached_study, name: 'Empty Annotation Study')
+  end
+
+  after(:all) do
+    @basic_study.destroy
+    @empty_study.destroy
+    @user.destroy
+    @user2.destroy
+  end
+
+  test 'methods should check view permissions' do
```

Comment:
I've noticed this test doesn't have a positive (e.g. `200`) assertion anywhere - we should add that.